lib/init/grass.py

```diff
@@ -742,7 +742,7 @@ def set_paths(grass_config_dir):
             s = p.stdout.read()
             p.wait()
             sys_man_path = s.strip()
-        except:
+        except Exception:
```

Comment:
There is no point in simply listing all the possible exceptions all the time. The exceptions can be reduced considering the expected exceptions. [os.devnull](https://docs.python.org/3/library/os.html#os.devnull) should give something useful unless Python is totally broken. We don't have to catch that, we probably want that to cause a traceback. This gets rid of the first three errors.

This piece of code is trying to get something from manpath, but ignore any failures - that is failures related to manpath (only). Number 4 and 5 look appropriate.

I don't know if the scenario for number 6 is likely. Is that something from the Popen doc (e.g., "if the object is not created properly, it will have just half of the attributes")? Would that apply to all other usages of attributes anywhere in the code?

Reply:
My guess is number 6 is unlikely. It is not always the case that AttributeError applies even though code assistance suggests it. I use ```subprocess``` errors here as follows. Please let me know if I am missing anything.
```python
except (subprocess.TimeoutExpired, subprocess.SubprocessError)
```

Reply:
These two are good.

Reply:
While working on this and another subprocess related piece of code, I read the documentation and it says:

> exception subprocess.TimeoutExpired
>
> Subclass of [SubprocessError](https://docs.python.org/3/library/subprocess.html#subprocess.SubprocessError), raised when a timeout expires while waiting for a child process.

So, only SubprocessError is needed from these two. However, the doc also says:

> The most common exception raised is [OSError](https://docs.python.org/3/library/exceptions.html#OSError). This occurs, for example, when trying to execute a non-existent file. Applications should prepare for [OSError](https://docs.python.org/3/library/exceptions.html#OSError) exceptions.

So, OSError is not limited to the cases listed above, so we should catch it.

This is what I now catch in #3694 (I will deal will merging the two when we get to it).